### PR TITLE
added a page that keeps track of added tracks

### PR DESCRIPTION
**Pull request by KimLinhVu**
What
===
* added a tab that displays list of tracks user previously added to their playlist
* store which track user added in db 

Why
===
* user can see which tracks they have added recently
* keep track of which tracks a user added in db

Who
===
@makushline @qibolun 

Test Plan
===
Screen Shot 2022-08-08 at 4 15 42 PM

### CODE REVIEW

File: server/routes/trackReceipt.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  } catch (error) {
+    next(error)
+  }
+})
+
+router.get('/', jwt.verifyJWT, async (req, res, next) => {
+  /* returns all track receipts belonging to current user */
+  try {
+    const userId = req.userId
+    const receipts = await TrackReceipt.find({ userId })
```
Comment (makushline) [inline server/routes/trackReceipt.js]: Nit: singular because find returns a single object.

File: server/routes/trackReceipt.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    /* delete expired receipts */
+    const promises = receipts.map(async (receipt) => {
+      const receiptDate = new Date(receipt.addedAt)
+      const receiptSeconds = receiptDate.getTime() / 1000
+      const timeNow = Date.now() / 1000
+
+      /* delete if longer than 7 days */
+      if ((timeNow - receiptSeconds) > 604800) {
```
Comment (makushline) [inline server/routes/trackReceipt.js]: 604800 is a [magic number](https://stackoverflow.com/questions/47882/what-is-a-magic-number-and-why-is-it-bad), move it into a well named variable.

File: server/models/TrackReceipt.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+const mongoose = require('mongoose')
+
+const trackReceiptSchema = new mongoose.Schema({
```
Comment (makushline) [inline server/models/TrackReceipt.js]: trackReceipt is not the right name for this object. This is more of an 'addedSong.'

**Review by makushline — CHANGES_REQUESTED** [review]
The PR generally looks good to me. Except trackReceipt is not the best name for this object, could you give the class a better name?

File: server/routes/trackReceipt.js (hunk 1 above)
Comment (KimLinhVu) [reply server/routes/trackReceipt.js]: This find returns an array of all the objects in the collection that has the specified userId.

**Review by makushline — APPROVED** [review]

**Pull request merged**